### DISCUSSION TITLE
상속은 왜 사용하는걸까?

### DISCUSSION THREAD

**Discussion by alstn113** [2024-woowacourse-study/level-interview#30] [OOP]
.

**Comment by jimi567** [comment]
상속을 사용하는 이유는 다음과 같다.

1. 이미 존재하는 클래스를 재사용할 수 있어, 효율적이고 중복 코드를 줄일 수 있다.

2. 확장성에 용이하고, 코드가 간결해진다.

3. 클래스 간의 계층 구조를 형성함으로써 코드의 구조를 명확하게 만들 수 있다.

4. 다형성을 활용할 수 있다. 부모 클래스로 묶어 같은 메소드 호출에 대해 각기 다른 동작을 수행할 수 있도록 해준다.

( [상속과 조합 비교](https://github.com/2024-woowacourse-study/level-interview/discussions/18)

**Comment by nayonsoso** [comment]
상속을 사용하면, 

1. 코드의 중복을 줄일 수 있습니다. 자식 클래스는 부모 클래스의 변수와 함수를 사용할 수 있기 때문입니다.
2. 또한 객체간의 관계가 IS-A 관계라는 것을 명시적으로 드러낼 수 있습니다.
3. 다형성을 활용하여 코드 변경을 줄일 수 있습니다. 자식 클래스가 어떻게 Overriding을 하는지에 따라 다른 기능을 할 수 있는데, 호출하는 쪽에서는 메서드의 내부 구현을 알 필요가 없습니다. 그저 호출만 해주면 됩니다. 따라서 추후 변경이 생기더라도, 메서드 내부만 수정해주면 되므로 코드의 변경이 줄어듭니다.

**Comment by le2sky** [comment]
중복 코드 제거 가능
변화에 대한 유연성, 확장성 증가
개발 시간 단축
is-a 관계를 명시적으로 표현
계층 구조를 형성
but, 상속을 적절히 사용했을 경우에만 해당.

**Comment by GIVEN53** [comment]
중복 코드 제거
확장성
계층 구조
다형성
is-a
